core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java

```diff
@@ -90,6 +90,14 @@ public StartDepositResponse deposit(Sep10Jwt token, StartDepositRequest request)
             .type(request.getType())
             .assetCode(request.getAssetCode())
             .assetIssuer(asset.getIssuer())
+            // NB: these are purposely set to incorrect values.
```

Comment:
Does it mean there will be a state of the transaction in DB when it has incorrect amounts?

Reply:
Yeah, the amounts are technically incorrect anyway since the Anchor updates fees asynchronously. Deposit instructions and withdraw anchor accounts (to be done in the next PR) will not be provided until the business server makes a request for funds RPC so there is no way for the user to accidentally submit the wrong amount.

Reply:
But if the user requests the transaction, won't we return incorrect data? Shouldn't amount just be null instead?

Reply:
Yes. the `NotifyAmountsUpdatedHandler` validation requires the assets to be set so it can't be null ([code](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyAmountsUpdatedHandler.java#L53-L67)). I guess we can remove the validation?

Reply:
I should probably think about this more, so I'll create a ticket to address this and merge it as is.

Reply:
I'm pretty sure RPC validation should just validate only the request having amount to not be null, if we expect transaction to start with null amounts

Reply:
Yeah, that makes sense. Created ANCHOR-525 to fix it.